Detectors/TPC/simulation/src/DigitTime.cxx

```diff
@@ -13,31 +13,28 @@
 /// \author Andi Mathis, TU München, andreas.mathis@ph.tum.de
 
 #include "TPCSimulation/DigitTime.h"
-#include "TPCSimulation/DigitRow.h"
-#include "TPCBase/Mapper.h"
 
 using namespace o2::TPC;
 
-void DigitTime::setDigit(size_t hitID, int cru, int row, int pad, float charge)
+void DigitTime::setDigit(int eventID, size_t hitID, CRU cru, GlobalPadNumber globalPad, float charge)
 {
-  /// Check whether the container at this spot already contains an entry
-  DigitRow *result = mRows[row].get();
-  if(result != nullptr) {
-    mRows[row]->setDigit(hitID, pad, charge);
-  }
-  else{
-    const Mapper& mapper = Mapper::instance();
-    mRows[row] = std::make_unique<DigitRow> (row, mapper.getPadRegionInfo(CRU(cru).region()).getPadsInRowRegion(row));
-    mRows[row]->setDigit(hitID, pad, charge);
-  }
-  mTotalChargeTimeBin+=charge;
+  mGlobalPads[globalPad].setDigit(eventID, hitID, charge);
+  const int gemStack = static_cast<int>(cru.gemStack());
+  mCommonMode[gemStack] += charge;
 }
 
 void DigitTime::fillOutputContainer(std::vector<o2::TPC::Digit> *output, o2::dataformats::MCTruthContainer<o2::MCCompLabel> &mcTruth,
-                                    std::vector<o2::TPC::DigitMCMetaData> *debug, int cru, int timeBin, float commonMode)
+                                    std::vector<o2::TPC::DigitMCMetaData> *debug, Sector sector, int timeBin)
 {
-  for(auto &aRow : mRows) {
-    if(aRow == nullptr) continue;
-    aRow->fillOutputContainer(output, mcTruth, debug, cru, timeBin, aRow->getRow(), commonMode);
+  const static Mapper &mapper = Mapper::instance();
+  GlobalPadNumber currentPad = 0;
+  for(auto &globalPad : mGlobalPads) {
+    /// Only actual signals are written to disk
+    if(globalPad.getChargePad() >0 ) {
+      /// \todo Obtain CRU from globalPadPos and Sector
+      int cru = sector.getSector();
+      globalPad.fillOutputContainer(output, mcTruth, debug, CRU(cru), timeBin, currentPad, getCommonMode(cru));
```

Comment:
This looks a bit funny, since the name is `cru`, but `getSector()` is called. I guess this is temporary as written in the comment.

Reply:
Yes, I could not find a mapping function to get the CRU from the Sector and the GlobalPadNumber